### PR TITLE
VIRTS-3315: Dependency Extractor

### PR DESCRIPTION
**Pull request by ArtificialErmine**
## Description

Implements a very basic dependency extractor for atomic abilities. Requires the atomic ability dependencies to be scripted in the original files; if not, the system will not expand the usual file produced to include dependency integration.

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Locally, validating the generated files as reasonable given the inputs.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [NA] I have made corresponding changes to the documentation
- [NA] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by ArtificialErmine** [comment]
Did some profiling. This version generates

**Comment by ArtificialErmine** [comment]
Current version (hoping to refine a bit more), can extract 117 dependencies from a total of 213 possible.

**Comment by ArtificialErmine** [comment]
Got coverage up to 179 out of 213 by tweaking a few things, but I'm not sure I'll be able to get many, if any, more. There's a sizeable number of tests that just tell the user to do something manually.

File: app/atomic_svc.py [hunk 1]
```diff
@@ -271,9 +276,25 @@ async def _prepare_executor(self, test, platform, executor):
...
         """
         payloads = []
-
-        command, payloads_command = await self._prepare_cmd(test, platform, executor, test['executor']['command'])
-        cleanup, payloads_cleanup = await self._prepare_cmd(test, platform, executor, test['executor'].get('cleanup_command', ''))
+        dep_construct = ""
+        if 'dependencies' in test:
+            for dependence in test['dependencies']:
+                try:
+                    test_exc = test.get('dependency_executor_name') \
```
Comment (uruwhy) [inline app/atomic_svc.py]: could this just be `test.get('dependency_executor_name', executor)`?

File: app/atomic_svc.py [hunk 2]
```diff
@@ -322,3 +343,71 @@ async def _save_ability(self, entries, test):
...
+            if 'exit 1' in segments[1]:
+                # check is "falsy"
+                output += f"{segments[0]}; then {prereq}; fi;"
+            else:
+                # check is "truthy"
+                output += f"{segments[0]}; then : ; else {prereq}; fi;"
+        elif prereq_type == 'psh':
+            if prereq_test.startswith('Try'):
+                temp = f"{prereq_test.replace('exit 1', prereq)}"
+                tstring = "Set-Variable -Name 'a' -Value 'pass'"
```
Comment (uruwhy) [inline app/atomic_svc.py]: is this just supposed to represent a no-op?

Comment (ArtificialErmine) [reply app/atomic_svc.py]: Yep - it's easier to do no-ops in some languages that others, but this was the one that came up for powershell in google. If you know of a shorter one, I'm game for making it more straightforward.

File: app/atomic_svc.py (hunk 1 above)
Comment (ArtificialErmine) [reply app/atomic_svc.py]: Yep, that seems to be the case. Good catch!

File: app/atomic_svc.py (hunk 2 above)
Comment (uruwhy) [reply app/atomic_svc.py]: You should be able to have just blank space between the braces, or even just a semicolon. Something like:
```
if ($true) { } else {echo hi}
if ($true) { ; } else {echo hi}
```

Comment (ArtificialErmine) [reply app/atomic_svc.py]: Nice, thanks for pointing that out.

**Review by uruwhy — APPROVED** [review]

**Pull request merged**